### PR TITLE
[15.0][4073][ADD] portal_user_login_email_change

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[4073](https://www.quartile.co/web#id=4073&cids=3&menu_id=517&model=project.task&view_type=form)

Related: https://github.com/qrtl/mi7-private/pull/168

### CODE REVIEW

File: portal_user_login_email_change/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        # Capture the old email for each partner before the update
+        old_emails = {partner.id: partner.email for partner in self}
+        result = super(ResPartner, self).write(vals)
+        if "email" in vals and self.env.user.has_group("base.group_portal"):
+            # Update user login if email has changed
+            for partner in self:
+                old_email = old_emails.get(partner.id)
+                new_email = vals.get("email")
+                if old_email != new_email:
+                    self._update_user_login(partner, old_email)
```
Comment (yostashiro) [inline portal_user_login_email_change/models/res_partner.py]:
```suggestion
        if "email" not in vals or not self.env.user.has_group("base.group_portal"):
            return super().write(vals)
        # Capture the old email for each partner before the update
        old_emails = {partner.id: partner.email for partner in self}
        result = super().write(vals)
        # Update user login if email has changed
        for partner in self:
            old_email = old_emails.get(partner.id)
            new_email = vals.get("email")
            if old_email != new_email:
                self._update_user_login(partner, old_email)
```

File: portal_user_login_email_change/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        return result
+
+    def _update_user_login(self, partner, old_email):
+        # Check if there is a related user with the same login as the old email
+        if partner.user_ids:
+            user = partner.user_ids.filtered(lambda u: u.login == old_email)
+            if user:
+                user.write({"login": partner.email})
```
Comment (yostashiro) [inline portal_user_login_email_change/models/res_partner.py:15]: Please move this above write().

**Review by yostashiro — APPROVED** [review]
Code review.

**Comment by yostashiro** [comment]
@kanda999 Please add the translation.

**Review by kanda999 — APPROVED** [review]

**Pull request merged**